src/gcp_scanner/credsdb.py

```diff
@@ -379,14 +378,38 @@ def creds_from_refresh_token(refresh_token_file):
   with open(refresh_token_file, encoding="utf-8") as f:
     creds_dict = json.load(f)
 
-  user_scopes = creds_dict.get("scopes", None)
-  if user_scopes is None:
-    user_scopes = ["https://www.googleapis.com/auth/cloud-platform"]
+  user_scopes = get_scope_from_refresh_token(creds_dict)
 
   return credentials.Credentials(
-      None,
-      refresh_token=creds_dict["refresh_token"],
-      token_uri=creds_dict["token_uri"],
-      client_id=creds_dict["client_id"],
-      client_secret=creds_dict["client_secret"],
-      scopes=user_scopes)
+    None,
+    refresh_token=creds_dict["refresh_token"],
+    token_uri=creds_dict["token_uri"],
+    client_id=creds_dict["client_id"],
+    client_secret=creds_dict["client_secret"],
+    scopes=user_scopes,
+  )
+
+
+def get_scope_from_refresh_token(context) -> Union[List[str], None]:
+  """The function is used to obtain scope from refresh token.
+
+  Args:
+    context: dictionary containing refresh_token data
+    Example:
+      {
+        "refresh_token": "<token>",
+        "client_id": "id",
+        "client_secret": "secret",
+      }
+  Returns:
+    a list of scopes
+  """
+  # Obtain access token from the refresh token
+  token_uri = "https://oauth2.googleapis.com/token"
+  context["grant_type"] = "refresh_token"
+  response = requests.post(token_uri, data=context)
+
+  # prepare the scope string into a list
+  raw = response.json().get("scope", None)
+  return raw.split(" ") if raw else None
```

Comment:
Hmm, ideally we need a test for the function or at least an issue so we do not forget about that.

Reply:
I think it is not easy without editor access to test project. Let's do issue at least.

Reply:
I can mock the API response and add a unit test. Later we can add another test using the test_porject in GCP.

Reply:
I implemented some unit tests by mocking the API response. 
As far as I read about unit testing. I think we should not depend on the `test_project` for unit tests. We can call those gcp-dependent tests **"Integration Test"**
I am not perfectly sure though as I don't have expert-level experience in these. Looking for your opinion here
@mshudrak